src/client/QXmppOutgoingClient.cpp

```diff
@@ -143,7 +143,7 @@ void QXmppOutgoingClientPrivate::connectToHost(const QString &host, quint16 port
 
     // override CA certificates if requested
     if (!config.caCertificates().isEmpty())
-        QSslConfiguration::defaultConfiguration().setCaCertificates(config.caCertificates());
+        q->socket()->sslConfiguration().setCaCertificates(config.caCertificates());
```

Comment:
AFAIK this won't work, because sslConfiguration() doesn't return a reference.

Reply:
Seems you are right there. Unfortunately I have some compilation issues with your suggestion but will push the proper fix once I found it.